src/dependency_parser/intra_patterns/github_up_down.py

```diff
@@ -0,0 +1,89 @@
+from src.dependency_parser.utils import steps, job_names
+
+
+def download_link(linked_dep, yml, job, action):
+    download_dep = action['with']['name']
+
+    # if the download dep still doesn't exist it is caused due to unknown artefact
+    if download_dep not in linked_dep.keys():
+        linked_dep[download_dep] = {
+            'producer': 'Unknown',
+            'consumer': [job],
+            'producer_inconsistency': [
+                {'warning': 'orange', 'problem': f"Artifacts {download_dep} not declared : "}
+            ]
+        }
+    elif 'needs' in yml['jobs'][job] and linked_dep[download_dep]['producer'] in yml['jobs'][job]['needs']:
+        linked_dep[download_dep]['consumer'] += [job]
+    # In case needs are missing.
+    else:
+        linked_dep[download_dep]['consumer'] += [job]
+        linked_dep[download_dep]['consumer_inconsistency'] = [
+            {'warning': 'orange', 'problem': "Missing Needs for job : " + linked_dep[download_dep]['producer']}
+        ]
+
+
+# Find out every upload pattern, and store the upload as producer job
+def upload_link(linked_dep, yml, job, action):
+    linked_dep[action['with']['name']] = {'producer': job, 'consumer': []}
+
+
+pattern = {
```

Comment:
Are these models or just the way we have divided up the model analysis?

Reply:
pas sur de comprnedre ce que ca veux dire en anglais, on a décidé de séparé la premiere fois qu'on a codé le upload du download, car ils sont différents